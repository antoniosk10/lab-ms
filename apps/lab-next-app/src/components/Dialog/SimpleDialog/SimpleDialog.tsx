import { Dialog, DialogTitle } from '@mui/material'
import { PropsWithChildren } from 'react'

type Props = {
  title: string
  isOpen: boolean
  onClose: () => void
}

export function SimpleDialog({
  children,
  title,
  isOpen,
  onClose,
}: PropsWithChildren<Props>) {
  return (
    <Dialog open={isOpen} onClose={onClose}>
      <DialogTitle align="center">{title}</DialogTitle>
      {children}
    </Dialog>
  )
}
