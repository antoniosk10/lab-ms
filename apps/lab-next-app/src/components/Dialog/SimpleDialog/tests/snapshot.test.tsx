import { render } from '@testing-library/react'

import { SimpleDialog } from '../SimpleDialog'

it('renders SimpleDialog unchanged', () => {
  const { baseElement } = render(
    <SimpleDialog isOpen={true} onClose={jest.fn} title="test" />
  )
  expect(baseElement).toMatchSnapshot()
})
