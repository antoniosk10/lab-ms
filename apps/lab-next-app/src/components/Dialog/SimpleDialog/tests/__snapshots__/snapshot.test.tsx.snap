// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AuthForm unchanged 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r0:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter MuiDialogTitle-root css-xrt2rx-MuiTypography-root-MuiDialogTitle-root"
          id=":r0:"
        >
          test
        </h2>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
